spec/jekyll_redirect_from/generator_spec.rb

```diff
@@ -94,55 +94,67 @@
     let(:redirects) { JSON.parse(contents) }
     let(:domain) { "http://jekyllrb.com" }
 
-    it "creates the redirets file" do
-      expect(path).to exist
-    end
-
-    it "contains redirects" do
-      expect(redirects.count).to eql(13)
+    context "when explicitly disabled" do
+      let(:site) do
+        fixture_site("site", { "redirect_from" => { "json" => false } })
```

Comment:
Where is this `fixture_site` method defined?

Reply:
This is at https://github.com/jekyll/jekyll-redirect-from/pull/207/files#diff-93830fa29d616f7c87903d08b5b1b29aR65, modeled after https://github.com/benbalter/jekyll-optional-front-matter/blob/master/spec/spec_helper.rb#L26-L35.

Reply:
Whoa! I did not see that file before.. :stuck_out_tongue: 
Additionally, there already seems to be a `fixture_path` [helper method](https://github.com/jekyll/jekyll-redirect-from/pull/207/files#diff-93830fa29d616f7c87903d08b5b1b29aL24).
Can that be reused..?

Reply:
Looks like it's only used by `fixture_site` (in [jekyll-optional-front-matter](https://github.com/benbalter/jekyll-optional-front-matter/blob/master/spec/spec_helper.rb#L26-L35) too), so happy to integrate its one-liner into `fixture_site` itself? Didn't want to drift from any existing paradigms in GitHub Pages-official plugins is all!

Reply:
The set ups are different. For example, there is no fixture directory named `"site"` in this repo or in this PR. So your spec is fundamentally flawed.

Reply:
Ah, quite fair. How is https://github.com/jekyll/jekyll-redirect-from/pull/207/commits/95d21757094313ee9f6f634bfb6e0e4b2dccf5a2 as an alternative? Afraid this is my first time with ruby/spec, but as best I can tell, this would seem to override `site.config`?

Reply:
Yes, that would override the `site.config`, but then it wouldn't quite be a proper configuration because the site's source points elsewhere. Here's what you need to do:
- In `spec/spec_helper.rb`, modify `site` to take an optional override:
```ruby
  def site(overrides = {})
    @site ||= Jekyll::Site.new(config.merge(overrides))
  end
```
- Use it in your test:
```ruby
let(:custom_site) { site("json" => false) }
...
```
*Note: The above is untested. However, you get the idea..*

Reply:
Hm, so with `site` redefined in that way, and a test like

```ruby
context "when explicitly disabled" do
  let(:custom_site) { site("redirect_from" => { "json" => false }) }
  it "does not create the redirects file" do
    puts custom_site.config.to_yaml
    ...
  end
end
```

it looks like `custom_site.config` ends up being identical to `site.config` (i.e., the overrides are missing), presumably because `site("redirect_from" => { "json" => false })` ends up returning `@site`, which is already set by some earlier invocation of `site()`, perhaps because of the outermost [`before`](https://github.com/jekyll/jekyll-redirect-from/blob/master/spec/jekyll_redirect_from/generator_spec.rb#L4-L9)? Am I perhaps misunderstanding what you had in mind?

Thanks!

Reply:
My bad.. The reason you're seeing this is because `:site` is a memoized method. It returns the same `@site` once the instance variable is set to a non-false, non-nil value.

Sorry abt that.

So I guess, the alternative route would be to:
```ruby
let(:custom_site) { Jekyll::Site.new(config.merge("json" => false)) }
...
```

Reply:
Ah, no worries. Would it be incorrect to override `site` itself, as with https://github.com/jekyll/jekyll-redirect-from/pull/207/commits/6c0f2791f1632b7f75982f977fede83c9b81a873? If a new symbol like `custom_site` is indeed necessary instead, is an additional `before` necessary (e.g., to call `custom_site.read`, `custom_site.generate`, `custom_site.render`, and/or `custom_site.write) or any other such setup?

Thanks!